e2e/tests/distribution_inspect.go

```diff
@@ -0,0 +1,201 @@
+			registry     string
+			authImageTag string
+		)
+		BeforeEach(func() {
+			// create a custom client to use http over unix sockets
+			uClient = client.NewClient(GetDockerHostUrl())
+			// get the docker api version that will be tested
+			version = GetDockerApiVersion()
+		})
+
+		BeforeEach(func() {
+			filename := "htpasswd"
+			// The htpasswd is generated by
+			// `<subject> run --entrypoint htpasswd public.ecr.aws/docker/library/httpd:2 -Bbn testUser testPassword`.
+			// We don't want to generate it on the fly because:
+			// 1. Pulling the httpd image can take a long time, sometimes even more 10 seconds.
+			// 2. It's unlikely that we will have to update this in the future.
+			// 3. It's not the thing we want to validate by the functional tests. We only want the output produced by it.
+			// Same password/logic from runfinch/common-tests
+			htpasswd := "testUser:$2y$05$wE0sj3r9O9K9q7R0MXcfPuIerl/06L1IsxXkCuUr3QZ8lHWwicIdS"
+			htpasswdDir := filepath.Dir(ffs.CreateTempFile(filename, htpasswd))
+			DeferCleanup(os.RemoveAll, htpasswdDir)
+			port := fnet.GetFreePort()
+			command.Run(opt, "run",
+				"-dp", fmt.Sprintf("%d:5000", port),
+				"--name", "registry",
+				"-v", fmt.Sprintf("%s:/auth", htpasswdDir),
+				"-e", "REGISTRY_AUTH=htpasswd",
+				"-e", "REGISTRY_AUTH_HTPASSWD_REALM=Registry Realm",
+				"-e", fmt.Sprintf("REGISTRY_AUTH_HTPASSWD_PATH=/auth/%s", filename),
+				registryImage)
+			registry = fmt.Sprintf(`localhost:%d`, port)
+			authImageTag = fmt.Sprintf(`%s/test-login:tag`, registry)
+			buildContext := ffs.CreateBuildContext(fmt.Sprintf(`FROM %s
+	CMD ["echo", "bar"]
+		`, defaultImage))
+			DeferCleanup(os.RemoveAll, buildContext)
+			command.Run(opt, "build", "-t", authImageTag, buildContext)
+		})
+
+		AfterEach(func() {
+			command.RemoveAll(opt)
+		})
+
+		It("should inspect distribution of alpine image", func() {
+			relativeUrl := client.ConvertToFinchUrl(version, fmt.Sprintf("/distribution/%s/json", newerAlpineImage))
+			res, err := uClient.Get(relativeUrl)
+			Expect(err).Should(BeNil())
+
+			Expect(res).To(HaveHTTPStatus(http.StatusOK))
+			var d registrytypes.DistributionInspect
+			err = json.NewDecoder(res.Body).Decode(&d)
+			Expect(err).Should(BeNil())
+			Expect(d.Descriptor).ShouldNot(BeNil())
+			Expect(d.Descriptor.MediaType).Should(Equal("application/vnd.oci.image.index.v1+json"))
+			// since the image is built and pushed on the test runner, the following aspects are OS/architecture dependent
+			// and not strictly checked
+			Expect(d.Descriptor.Digest).Should(MatchRegexp(`sha256:\w{64}`))
+			Expect(d.Descriptor.Size).Should(BeNumerically(">", 0))
+			Expect(d.Descriptor.URLs).Should(HaveLen(0))
+			Expect(d.Descriptor.Data).Should(BeNil())
+			Expect(d.Descriptor.Platform).Should(BeNil())
+			Expect(d.Descriptor.ArtifactType).Should(Equal(""))
+			Expect(d.Platforms).ShouldNot(BeNil())
+		})
+
+		It("should inspect distribution of older alpine image", func() {
```

Comment:
What is the difference between an old and new alpine image in terms of distribution spec?

Reply:
Like @Shubhranshu153 alluded in his comment (https://github.com/runfinch/finch-daemon/pull/121#discussion_r1894239698), this is to test the two different media types.

The newer alpine image uses `application/vnd.oci.image.index.v1+json` and the older image uses `application/vnd.docker.distribution.manifest.list.v2+json`